backend/app/api/v1/commons/utils.py

```diff
@@ -65,3 +67,28 @@ def getReleaseStream(row):
     elif row["releaseStream"].__contains__("ec"):
         return "Engineering Candidate"
     return "Stable"
+
+
+def build_sort_terms(sort_string: str) -> list[dict[str, str]]:
+    """
+
+    Validates and transforms a sort string in the format 'sort=key:direction' to
+    a list of dictionaries [{key: {"order": direction}}].
+
+    :param sort_string: str, input string in the format 'sort=key:direction'
+
+    :return: list, transformed sort structure or raises a ValueError for invalid input
+
+    """
+
+    pattern = r"^([\w]+):(asc|desc)$"
```

Comment:
You don't need the square brackets for a single symbol. `r"^(\w+):(asc|desc)$"` is the same thing. As long as all columns are just letters, digits, and "_", `\w+` will work fine. Is that true? (For example, if we need to handle a hyphen, that would be `[\w-]+`.)

And since you're not doing an pre-validation of the field names, you'll get an Elasticsearch error if a bad field is passed ... is that OK? Opensearch will return an error document rather than the list of hits, and I think that'll break the search service's `post` rather badly.

Reply:
Updated the code to validate the field-names and directions